repositories/memory/topics/get_topic_test.go

```diff
@@ -0,0 +1,88 @@
+package topics
+
+import (
+	"context"
+	"testing"
+
+	"github.com/matheusmosca/walrus/domain/entities"
+	"github.com/matheusmosca/walrus/domain/vos"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetTopic(t *testing.T) {
+	type args struct {
+		topicName vos.TopicName
+	}
+	type testCase struct {
+		description string
+		args        args
+		beforeRun   func(storage map[vos.TopicName]entities.Topic)
+	}
+
+	// positiveTestcase creates a topic and stores it in the repository.storage as a part of its beforeRun routine
+	// hence these test cases should run with positive results for GetTopic method
+	positiveTestcase := []testCase{
```

Comment:
Why do you created 2 slices within the same function test? If you wanted to separe the success and failure tests, it would be better to create 2 Test func like `TestGetTopic_Sucess` and `TestGetTopic_Failure`. Then each one would have its own slice. WDYT @amoghrajesh?

Reply:
@matheusmosca I think this is a good idea. Will make the success and failure cases separate and modular.